### PR TITLE
ESLint, Cleanup

### PR DESCRIPTION
**Pull request by bkendall**
Though this looks like a large change, the quick summary is that it's a lot of cleanup, and not a lot of functional changes:

- Use `eslint` instead of `jshint`. New config is in place, and `npm run lint` is correct
- Removed `lodash`! Functionality has been replaced by https://github.com/tjmehta/101 functions
- Added testing for `iojs`. Since it's around, we may as well support it... though I think it will only be on 'best effort' from our side...
- Cleaned up the README a little as well

### CODE REVIEW

**Comment by bkendall** [comment]
@tjmehta I think I used the appropriate `101` things in replacing `lodash`. Could you take a quick once-over and make sure I did? The big one I had to change out was being able to just iterate over all the key-value pairs in an object (just went with `Object.keys(...).map(...)`, since there's nothing I can think of that will easily do that in `101` - maybe an idea?).

File: lib/middleware.js [hunk 1]
```diff
@@ -212,13 +219,13 @@ var middlewares = module.exports = {
     container.NetworkSettings = {
       Bridge: 'docker0',
       Gateway: '172.17.42.1',
-      IPAddress: '172.17.0.'+newPort(),
+      IPAddress: '172.17.0.' + newPort(),
```
Comment (podviaznikov) [inline lib/middleware.js]: strange that we are calling `newPort()` here. But I can see that it works

**Comment by podviaznikov** [comment]
looks good to me

File: lib/middleware.js (hunk 1 above)
Comment (bkendall) [inline lib/middleware.js]: huh - weird

**Pull request merged**